src/main/scala/chisel3/simulator/scalatest/HasCliOptions.scala

```diff
@@ -162,6 +162,15 @@ trait HasCliOptions extends HasConfigMap { this: TestSuite =>
     options += option.name -> option
   }
 
+  final def overrideOption(option: CliOption[_]): Unit = {
+    if (!options.contains(option.name))
+      throw new Exception(
+        s"unable to override existing option with name '${option.name}' because this option was not previously defined"
+      )
+
+    options += option.name -> option
```

Comment:
```suggestion
    options(option.name) = option
```
I think using the setter makes the intent here more obvious (even though you have already done the check).